tests/error_message.rs

```diff
@@ -0,0 +1,17 @@
+extern crate snafu;
+
+use std::error::Error as _;
+use snafu::{Snafu, ResultExt};
+
+
+#[derive(Debug, Snafu)]
+enum Error {
+    BadId{ source: std::io::Error },
+}
+
+#[test]
+fn test_error_message() {
+    let err : Error = std::fs::read("/non/existent/file--test").eager_context(BadId{}).unwrap_err();
```

Comment:
I don't think `fs::read` exists in Rust 1.18, but you can use `File::open` for the same effect.